### ISSUE TITLE
Database versioning

### ISSUE THREAD

**Issue by dibley1973** [dibley1973/OpenRMS#39]
As we are intending for Open RMS to be available for more than one client, I'd like to propose moving to database versioning and script control more like Vlad describes [here](http://enterprisecraftsmanship.com/2015/08/10/database-versioning-best-practices/).

Please discuss.

**Comment by jadjare** [comment]
I'm assuming openRMS will be delivered as a SaaS product, rather than client installs.

Read the first page of Vlad's post, all options seem a nightmare.  Because DBs aren't compiled code, they always run the risk of getting out of sync... Does anyone know if dB versioning issues disappear when using a NoSql store?

**Comment by dibley1973** [comment]
Yes SaaS is what we are aiming for, but I have been wondering if rather than just one big DB with multiple tenants, in the age of security breaches and such like, I am feeling multiple databases, which one for each tenant. Same for the web site.

Then if a client does not want an upgrade they can stay on a lower version. I will read up on what you mentioned at lunchtime today if I get chance, as only just seen your comment here and need to get ready for work now!

**Comment by dibley1973** [comment]
Out of interest, a similar way of versioning was what _Infospeed_ did for their client DBs and code bases.
